src/net/connection/connection_pool.rs

```diff
@@ -74,9 +82,9 @@ impl ConnectionPool {
         Ok(thread::Builder::new()
             .name("check_for_timeouts".into())
             .spawn(move || loop {
-                match connections.read() {
-                    Ok(lock) => {
-                        ConnectionPool::check_for_timeouts(&*lock, poll_interval, &events_sender);
+                match connections.write() {
```

Comment:
You may use a read if no connection has timed out: returns the vec of timed out addr and get the read guard only if the list is not empty. What do you think?

Reply:
Yes, there should be a read and only a write lock once you know you need to remove. There is already code for this btw.